lib/tasks/add_covid_stamp.rake

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+namespace :covid_stamp do
+  task add: :environment do
+    ActiveRecord::Base.transaction do
+      instrument = Instrument.find_or_create_by!(name: 'Beckman NX', barcode: '009851')
+      instrument_process = InstrumentProcess.create!(name: 'COVID stamp', key: 'covid_stamp')
+      InstrumentProcessesInstrument.create!(
+        instrument: instrument,
+        instrument_process: instrument_process,
+        bed_verification_type: 'Verification::DilutionPlate::Nx'
+      )
+    end
+  end
+
+  task remove: :environment do
+    ActiveRecord::Base.transaction do
+      instrument_process = InstrumentProcess.find_by_key('covid_stamp')
+      instrument_processes_instruments = instrument_process.instrument_processes_instruments
+      instrument_processes_instruments.each(&:destroy!)
+      instrument_process.destroy!
+    end
+  end
+end
```

Comment:
Just curious, why we are not using a migration to apply this changes on deployment?

Reply:
I'm not sure what the preferred way is as a team of doing data migrations. I guess options are:
1) run script in console
2) run rake task in console
3) migration
4) something else

James preferred a rake task for reasons like - the migration might be dependent on data that already exists in live, but not in dev environments - for instance, I initially had the first line as `instrument = Instrument.find_by!(name: 'Beckman NX')` but then realised that migration would fail in a dev setup.

He also mentioned something called RecordLoader that he's been working on in SS.

Sounds to me like might be something to have a wider team discussion about when Heron stuff quietens down again?

Reply:
Yes, generally adding data in migration is recommended against, and instead migrations are intended to be primarily concerned with your schema. (Migrations that change the data to support a new schema are fine, but they should make as few assumptions about the underlying content as possible.)

In addition to the concerns Katy mentioned, migrations have the limitation that once they've run, and have been recorded in the schema, they wont run again. This means that if you run something like `rake db:setup`, your records wont exist, and there is no easy way of recreating them if you want them.

The RecordLoader I've been playing around with is a way of generating data in an idempotent manner, which can then be triggered in a rake task, or via the seeds. As an added benefit, it can even have tests written for it, something which is tricky for migrations.

@emrojo In terms of triggering all this automatically, the intent is that you'll just be able to set the rake tasks up to run after a shared on deployment task. I think one of these already exists, but I need to check psd-deployment. I'm also hoping to add a bit of functionality to handle wip stuff, that way we can merge stuff in early, and trigger it manually as an when we want.